src/controllers/portfolio/portfolio.ts

```diff
@@ -52,6 +52,19 @@ export class PortfolioController extends EventEmitter {
 
   pending: PortfolioControllerState
 
+  // A queue to prevent race conditions when calling `updateSelectedAccount`.
+  // All calls are queued by network and account.
+  // Each time `updateSelectedAccount` is invoked to update the latest or pending state, the call is added to the queue.
+  // If a previous call is still running, the new call will be queued and executed only after the first one completes,
+  // regardless of whether it succeeds or fails.
+  // Before implementing this queue, multiple `updateSelectedAccount` calls made in a short period of time could cause
+  // the response of the latest call to be overwritten by a slower previous call.
+  queue: {
```

Comment:
rename to `#queue` because it will only be used internally by the ctrl

Reply:
I planned it in that way, but forgot to change its scope. Good point!